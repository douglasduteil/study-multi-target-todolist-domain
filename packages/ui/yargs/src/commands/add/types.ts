//

import { Arguments } from "yargs";

//

export type AddArguments = Arguments<Options>;
export type Aliases = "a";

export interface Options {
  message: string;
}
