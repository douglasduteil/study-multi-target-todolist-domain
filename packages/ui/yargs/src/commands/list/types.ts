//

import { Arguments } from "yargs";

//

export type ListArguments = Arguments<{ [key in keyof ListOptions]: boolean }>;
export type Aliases = "ls" | "la" | "ll";

export interface ListOptions {
  all: boolean;
  // XXX(douglasduteil): Move json out of here
  // json should be part of the "list" filter as it's format
  json: boolean;
  long: boolean;
}
