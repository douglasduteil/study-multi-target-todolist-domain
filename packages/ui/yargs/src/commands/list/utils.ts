//

import { ALIASES_DEFAULT_OPTIONS } from "./options";
import { Aliases, ListArguments } from "./types";

//

export function mapAliasOnOptions(argv: ListArguments): ListArguments {
  const alias = argv._[0] as Aliases;
  const options = ALIASES_DEFAULT_OPTIONS[alias];
  return { ...argv, ...options };
}
