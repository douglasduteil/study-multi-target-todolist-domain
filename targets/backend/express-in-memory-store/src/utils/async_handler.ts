//

import { NextFunction, Request, RequestHandler, Response } from "express";

//

// Inspired by https://www.npmjs.com/package/express-async-handler
export const asyncMiddleware = (fn: RequestHandler) => (
  req: Request,
  res: Response,
  next: NextFunction
) => {
  Promise.resolve(fn(req, res, next)).catch(next);
};
