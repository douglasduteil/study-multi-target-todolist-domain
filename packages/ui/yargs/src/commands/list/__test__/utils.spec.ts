//

import { ListOptions } from "../types";
import { mapAliasOnOptions } from "../utils";

//

const defaultArgv: ListOptions = { all: false, json: false, long: false };

//

test("should do nothing if no alias is given", () => {
  expect(
    mapAliasOnOptions({ $0: "foo", _: [], ...defaultArgv })
  ).toMatchSnapshot();
});

test("should do nothing if the alias is not found", () => {
  expect(
    mapAliasOnOptions({
      $0: "foo",
      _: ["oof"],
      ...defaultArgv
    })
  ).toMatchSnapshot();
});

test('"ls" do nothing', () => {
  expect(
    mapAliasOnOptions({
      $0: "foo",
      _: ["ls"],
      ...defaultArgv
    })
  ).toMatchSnapshot();
});

test('"la" should activate "all" option', () => {
  expect(
    mapAliasOnOptions({
      $0: "foo",
      _: ["la"],
      ...defaultArgv
    })
  ).toMatchSnapshot();
});

test('"ll" should activate "long" option', () => {
  expect(
    mapAliasOnOptions({
      $0: "foo",
      _: ["ll"],
      ...defaultArgv
    })
  ).toMatchSnapshot();
});
