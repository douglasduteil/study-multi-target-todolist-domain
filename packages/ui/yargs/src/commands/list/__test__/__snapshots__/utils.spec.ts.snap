// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"la" should activate "all" option 1`] = `
Object {
  "$0": "foo",
  "_": Array [
    "la",
  ],
  "all": true,
  "json": false,
  "long": false,
}
`;

exports[`"ll" should activate "long" option 1`] = `
Object {
  "$0": "foo",
  "_": Array [
    "ll",
  ],
  "all": false,
  "json": false,
  "long": true,
}
`;

exports[`"ls" do nothing 1`] = `
Object {
  "$0": "foo",
  "_": Array [
    "ls",
  ],
  "all": false,
  "json": false,
  "long": false,
}
`;

exports[`should do nothing if no alias is given 1`] = `
Object {
  "$0": "foo",
  "_": Array [],
  "all": false,
  "json": false,
  "long": false,
}
`;

exports[`should do nothing if the alias is not found 1`] = `
Object {
  "$0": "foo",
  "_": Array [
    "oof",
  ],
  "all": false,
  "json": false,
  "long": false,
}
`;
